flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/ConfigMapDecorator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.decorators;
+
+import org.apache.flink.client.cli.CliFrontend;
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.kubernetes.configuration.KubernetesConfigOptions;
+import org.apache.flink.kubernetes.kubeclient.resources.FlinkConfigMap;
+import org.apache.flink.kubernetes.utils.KubernetesUtils;
+import org.apache.flink.util.Preconditions;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.configuration.GlobalConfiguration.FLINK_CONF_FILENAME;
+import static org.apache.flink.kubernetes.utils.Constants.CONFIG_FILE_LOG4J_NAME;
+import static org.apache.flink.kubernetes.utils.Constants.CONFIG_FILE_LOGBACK_NAME;
+
+/**
+ * Decorate config map for flink configuration and log files.
+ */
+public class ConfigMapDecorator extends Decorator<ConfigMap, FlinkConfigMap> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ConfigMapDecorator.class);
+
+	@Override
+	protected ConfigMap doDecorate(ConfigMap resource, Configuration flinkConfig) {
+
+		String confDir = CliFrontend.getConfigurationDirectoryFromEnv();
+		Preconditions.checkArgument(confDir != null);
+
+		StringBuilder flinkConfContent = new StringBuilder();
+		Map<String, String> flinkConfigMap = new HashMap<>(flinkConfig.toMap());
+		// Remove some keys should not be taken to jobmanager and taskmanager.
+		flinkConfigMap.remove(ConfigConstants.ENV_FLINK_PLUGINS_DIR);
```

Comment:
Not sure why we should exclude this config, could you elaborate it a bit?

Reply:
The code base has changed and `ConfigConstants.ENV_FLINK_PLUGINS_DIR` will not be set in flink configuration. So i will remove this line.
For `KubernetesConfigOptions.KUBE_CONFIG_FILE`, it is the kube config for client only. And it should not be set to jobmanager and taskmanager pod. The file does not exist there.

Reply:
Thanks for the explanation.